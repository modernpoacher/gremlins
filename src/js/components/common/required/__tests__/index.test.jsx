import React from 'react'
import {
  snapshotOf
} from 'react-component-snapshot'

import '@testing-library/jest-dom'

import {
  render
} from '@testing-library/react'

import Required from '#gremlins/components/common/required'

describe('#gremlins/components/common/required', () => {
  describe('<Required />', () => {
    it('renders', () => {
      const {
        container: {
          firstElementChild: required
        }
      } = render(
        <Required />
      )

      expect(required)
        .toBeInstanceOf(HTMLSpanElement)
    })

    it('matches the snapshot', () => {
      const {
        container: {
          firstElementChild: required
        }
      } = render(
        <Required />
      )

      expect(snapshotOf(required))
        .toMatchSnapshot()
    })
  })
})
