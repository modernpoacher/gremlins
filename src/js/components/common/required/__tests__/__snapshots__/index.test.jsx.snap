// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#gremlins/components/common/required <Required /> matches the snapshot 1`] = `
<span
  className="is-required"
>
  *
</span>
`;
